ocaml/xapi/xapi.ml

```diff
@@ -675,8 +675,16 @@ let check_network_reset () =
               | None ->
                   phy_pif
             in
-            Xapi_pif.reconfigure_ip ~__context ~self:pif ~mode ~iP ~netmask
-              ~gateway ~dNS ;
+            (* Reconfigure accordingly to primary_address_type *)
+            ( match Db.PIF.get_primary_address_type ~__context ~self:pif with
+            | `IPv4 ->
+                Xapi_pif.reconfigure_ip ~__context ~self:pif ~mode ~iP ~netmask
+                  ~gateway ~dNS
+            | `IPv6 ->
+                Xapi_pif.reconfigure_ipv6 ~__context ~self:pif ~mode
+                  ~iPv6:(iP ^ "/" ^ netmask)
+                  ~gateway ~dNS
```

Comment:
I think it would be better to add the netmask parameter to the call, just like the ipv4 version

Reply:
The thing is, the `IPv6` field of a PIF is a list of `ip6/cidr` string.
So it'd mean maintaining 2 lists and knowing which cidr goes with which ip.